yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -23,20 +31,46 @@ export class AztecKVTxPool implements TxPool {
   /** Index from tx priority (stored as hex) to its tx hash, filtered by pending txs. */
   #pendingTxPriorityToHash: AztecMultiMap<string, string>;
 
+  /** Archived txs map for future lookup. */
+  #archivedTxs: AztecMapWithSize<string, Buffer>;
+
+  /** Indexes of the archived txs by insertion order. */
+  #archivedTxIndices: AztecMap<number, string>;
+
+  /** Index of the most recently inserted archived tx. */
+  #archivedTxHead: AztecSingleton<number>;
+
+  /** Index of the oldest archived tx. */
+  #archivedTxTail: AztecSingleton<number>;
+
+  /** Number of txs to archive. */
+  #archivedTxLimit: number;
+
   #log: Logger;
 
   #metrics: PoolInstrumentation<Tx>;
 
   /**
-   * Class constructor for in-memory TxPool. Initiates our transaction pool as a JS Map.
+   * Class constructor for KV TxPool. Initiates our transaction pool as an AztecMap.
    * @param store - A KV store.
    * @param log - A logger.
    */
-  constructor(store: AztecKVStore, telemetry: TelemetryClient, log = createLogger('p2p:tx_pool')) {
+  constructor(
+    store: AztecKVStore,
+    telemetry: TelemetryClient,
+    archivedTxLimit = getP2PConfigFromEnv().archivedTxLimit,
```

Comment:
Let's avoid reaching out to env within the constructor of a component. Instead, we should inject this value from the corresponding factory method. See `new AztecKVTxPool` in `p2p/src/client/factory.ts`, where you have a `P2PConfig` object available and you can pass the `archivedTxLimit` from there.